torch/nn/parallel/distributed.py

```diff
@@ -1138,15 +1138,12 @@ def join(
                         # buffers in the forward pass.
                         self._check_and_sync_module_buffers()
 
-                        (
-                            work,
-                            should_sync_backwards_tensor,
-                        ) = self._check_global_requires_backward_grad_sync(
+                        work = self._check_global_requires_backward_grad_sync(
                             is_joined_rank=True
                         )
                         work.wait()
                         # If nonzero, then we should sync in the bwd pass.
-                        should_sync_backwards = should_sync_backwards_tensor.item() != 0
+                        should_sync_backwards = work.result()[0].item() != 0
```

Comment:
Is there any possibility that `work` here does not call `wait()` first?

Reply:
In general, yes, but we call `wait()` in the line right above, so it should not be an issue for this case?